src/patterns/WordChart/WordChart.vue

```diff
@@ -0,0 +1,178 @@
+</template>
+
+<script>
+import _filter from 'lodash/filter'
+import _orderBy from 'lodash/orderBy'
+
+import tokens from '../../utils/tokens'
+
+import IGNORED_WORDS_ENGLISH from './ignoredWords/en'
+import IGNORED_WORDS_SPANISH from './ignoredWords/es'
+import IGNORED_WORDS_PORTUGUESE from './ignoredWords/pt'
+
+const WHITESPACE = /\s+/
+const NOT_ALPHANUMERIC_UNDERSCORE_DASH = /[^\w\-]/gi
+
+/**
+ * Draws a chart indicating incidence of words
+ *
+ * **Events**
+ *
+ * `@selectWord`
+ * emits object `{ word:String, count:Number }`
+ */
+export default {
+  name: 'WordChart',
+  status: 'prototype',
+  release: '0.1.0',
+  model: {
+    prop: 'active',
+  },
+  props: {
+    /**
+     * The body of text word incidence will be measured in
+     */
+    text: {
+      type: String,
+      required: true,
+    },
+    /**
+     * Removes non-meaningful words for language. Currently supports `en`, `es`, and `pt`
+     */
+    language: {
+      type: String,
+      default: 'en',
+    },
+    /**
+     * The html element used for the word chart container
+     */
+    type: {
+      type: String,
+      default: 'div',
+    },
+    /**
+     * The font-size of the largest word in the chart, all other words will be sized proportionally by their incidence
+     */
+    maximumFontSize: {
+      type: Number,
+      default: 5,
+    },
+    /**
+     * The unit of maximumFontSize
+     */
+    maximumFontSizeUnits: {
+      type: String,
+      default: 'rem',
+    },
+    /**
+     * The minimum incidence a word must have to be displayed
+     */
+    minWordCount: {
+      type: Number,
+      default: 5,
+    },
+    /**
+     * The minimum number of characters a word must be to be displayed
+     */
+    minWordLength: {
+      type: Number,
+      default: 2,
+    },
+    /**
+     * The maximum number of words in the chart
+     */
+    maxWordsShown: {
+      type: Number,
+      default: 50,
+    },
+  },
+  data() {
+    return {
+      colors: tokens.primaryColors.concat(tokens.secondaryColors).map(c => c.value),
+    }
+  },
+  computed: {
+    ignoredWords() {
+      return {
+        en: IGNORED_WORDS_ENGLISH,
+        es: IGNORED_WORDS_SPANISH,
+        pt: IGNORED_WORDS_PORTUGUESE,
+      }[this.language]
+    },
+    words() {
+      // returned data structure
+      // [
+      //   {
+      //     text: 'amazing',
+      //     color: 'rgb(120, 200, 100)',
+      //     remSize: 3,
+      //   },
+      // ]
+      const words = this.text.split(WHITESPACE)
+      let maxCount = 0
+
+      const treatWord = word => word.toLowerCase().replace(NOT_ALPHANUMERIC_UNDERSCORE_DASH, '')
+
+      const countReducer = (acc, w) => {
+        const word = treatWord(w)
+        
+        const skipWord = word.length < this.minWordLength || this.ignoredWords.includes(word)
+        if (skipWord) return acc
+
+        const count = (acc[word] || 0) + 1
+        if (count > maxCount) maxCount = count
+
+        return { ...acc, [word]: count }
```

Comment:
[nb] also for performance reasons, probably faster to mutate the `acc` object rather than copy it every time.
```suggestion
        acc[word] = count
        return acc
```

Reply:
Probably. I've spent too much of my js life writing immutable js 😅